### ISSUE TITLE
Cannot move to next column if widget is not on row 0

### ISSUE THREAD

**Issue by aaronpierce** [jwlodek/py_cui#35]
Getting an issue where I cannot move to the next column if there is not an item on the first row of the next column. The code im using as follows (with offending imports commented out.):
```
import py_cui
#import pantrack

class Interface():

	def __init__(self):
		#self.db = pantrack.Database()
		self.master = py_cui.PyCUI(6,2)
		self.task_menu = self.master.add_scroll_menu('Tasks', 0, 0, row_span=6)
		self.update_button = self.master.add_button('Update', 1, 1, command=self.update_tasks)

	def start(self):
		self.master.start()

	def update_tasks(self):
		#tasks = [task['short_desc'] for task in self.db.get_tasks()]
		self.task_menu.add_item_list(tasks)

if __name__ == '__main__':
        ui = Interface()
        ui.start()

```
The error still persisted when i tried:
- Different types of widgets
- The first column widget didnt use the max `row_span` 
- The `row` of the first column widget was greater than 0.

**Comment by jwlodek** [comment]
Yeah this is an issue I was aware of. At the moment, the widgets only search directly in the column/row they are in. The logic for assigning neighbors should probably be looked at. I was thinking about storing widget ids for up, down, left, and right, as they are added in, so that it doesn't have to calculate this on the fly, and so that better matches are assigned for each direction if available.

**Comment by aaronpierce** [comment]
I think I have an idea to resolve after looking around how it works now. Ill play around with some stuff and see if its something you want to use. If its not, no biggie!

**Comment by jwlodek** [comment]
If you put something together I'll take a look for sure. I'll eventually do something myself, but at the moment not much time for personal project work, so if you solve the problem all the better for me!

**Comment by jwlodek** [comment]
Closing, resolved with v0.1.0. Some progress can still be made, as diagonal movements are still not supported, so widgets still need to have some common border, but I think that can be a separate issue.
